CMakeLists.txt

```diff
@@ -289,33 +289,51 @@ endif(NOT NOT_NEED_LIBNSL)
 
 check_function_exists(gethostname HAVE_GETHOSTNAME)
 
-set(OPENSSL_DEFAULT ON)
 if(WIN32)
-  set(OPENSSL_DEFAULT OFF)
   check_library_exists_concat("ws2_32" getch        HAVE_LIBWS2_32)
   check_library_exists_concat("winmm"  getch        HAVE_LIBWINMM)
 endif()
 
-option(CMAKE_USE_OPENSSL "Use OpenSSL code. Experimental" ${OPENSSL_DEFAULT})
-mark_as_advanced(CMAKE_USE_OPENSSL)
+# check SSL libraries
+# TODO support GNUTLS, NSS, POLARSSL, AXTLS, CYASSL
 
 if(WIN32)
-  CMAKE_DEPENDENT_OPTION(CURL_WINDOWS_SSPI "Use windows libraries to allow NTLM authentication without openssl" ON
-    "NOT CMAKE_USE_OPENSSL" OFF)
-  mark_as_advanced(CURL_WINDOWS_SSPI)
+  option(CMAKE_USE_WINSSL "enable Windows native SSL/TLS" OFF)
+  if(CMAKE_USE_WINSSL AND NOT SSL_ENABLED)
+    set(SSL_ENABLED ON)
+    set(USE_SCHANNEL ON) # Windows native SSL/TLS support
+    set(USE_WINDOWS_SSPI ON) # CMAKE_USE_WINSSL implies CURL_WINDOWS_SSPI
+    list(APPEND CURL_LIBS "crypt32")
+  endif()
+
+  cmake_dependent_option(CURL_WINDOWS_SSPI "Use windows libraries to allow NTLM authentication without openssl" ON
+    CMAKE_USE_WINSSL OFF)
+  if(CURL_WINDOWS_SSPI)
+    set(USE_WINDOWS_SSPI ON)
+  endif()
 endif()
 
-set(USE_OPENSSL OFF)
-set(HAVE_LIBCRYPTO OFF)
-set(HAVE_LIBSSL OFF)
+if(APPLE)
+  option(CMAKE_USE_DARWINSSL "enable Apple OS native SSL/TLS" OFF)
+  if(CMAKE_USE_DARWINSSL AND NOT SSL_ENABLED)
+    find_library(SECURITY_FRAMEWORK "Security")
+    if(SECURITY_FRAMEWORK)
+      set(SSL_ENABLED ON)
+      set(USE_DARWINSSL ON)
+      list(APPEND CURL_LIBS "-framework CoreFoundation -framework Security")
```

Comment:
Any library that begins with `-` gets passed straight through to the command-line, so it works as two options are as one option. After taking another look at this code, I replaced this with `"${COREFOUNDATION_FRAMEWORK}" "${SECURITY_FRAMEWORK}"`, since it seems more correct (and it already goes to the trouble of finding Security.framework, so might as well...)